src/validation/difficulty.cairo

```diff
@@ -125,11 +120,13 @@ fn bits_to_target(bits: u32) -> Result<u256, ByteArray> {
         return Result::Ok(target);
     } else if exponent <= 3 {
         // For exponents 1, 2, and 3, divide by 256^(3 - exponent) i.e right shift
-        let shift = 8 * (3 - exponent);
-        target = shr(target, shift);
+        for _ in 0..(3 - exponent) {
+            target /= 0x100;
+        }
     } else {
-        let shift = 8 * (exponent - 3);
-        target = shl(target, shift);
+        for _ in 0..(exponent - 3) {
+            target *= 0x100;
```

Comment:
Since there are only 3 cases, its wiser to match the expononent value and use a constant rather than the loop + consecutive divisions or multiplications, especially for u256 type.